content/guides/code-splitting.md

```diff
@@ -1,41 +1,174 @@
 
-A typical application can depend on many third party libraries for framework/functionality needs. Unlike application code, code present in these libraries does not change often.
+__webpack.config.js__
 
-If we keep code from these libraries in its own bundle, separate from the application code, we can leverage the browser's caching mechanism to cache these files for longer durations on the end user's machine.
+``` js
+const path = require('path');
 
-For this to work, the `[hash]` portion in the vendor filename must remain constant, regardless of application code changes. Learn [how to split vendor/library](/guides/code-splitting-libraries) code using the `CommonsChunkPlugin`.
+module.exports = {
+  entry: {
+    index: './src/index.js',
+    vendor: [
+      'lodash'
+    ]
+  },
+  output: {
+    filename: '[name].bundle.js',
+    path: path.resolve(__dirname, 'dist')
+  }
+};
+```
 
-### CSS Splitting
+This will yield the following build result:
 
-You might also want to split your styles into a separate bundle, independent from application logic.
-This enhances cacheability of your styles and allows the browser to load the styles in-parallel with your application code, thus preventing a FOUC ([flash of unstyled content](https://en.wikipedia.org/wiki/Flash_of_unstyled_content)).
+?> Add bash example of webpack output
 
-Learn [how to split CSS](/guides/code-splitting-css) using the `ExtractTextWebpackPlugin`.
+As mentioned there are some pitfalls to this approach:
 
+- If there are any duplicated modules between entry chunks they will be included in both bundles.
+- It isn't as flexible and can't be used to dynamically split code with the core application logic.
 
-## On Demand Code Splitting
+The first of these two points is definitely an issue for our example, as `moment` is also imported within `./src/index.js` and will thus be duplicated in the output. See the `CommonChunkPlugin` example below for a solution to this problem.
 
-While resource splitting of the previous kind requires the user to specify the split points upfront in the configuration, one can also create dynamic split points in the application code.
 
-This can be used for more granular chunking of code, for example, per our application routes or as per predicted user behaviour. This allows the user to load non-essential assets on demand.
+## Plugins & Loaders
 
-Learn [how to split on demand](/guides/code-splitting-async) using `import()` or `require.ensure()`.
+There are multiple plugins and loaders in webpack's ecosystem that can help with splitting and managing split code. The most widely utilized of these is the [`CommonsChunkPlugin`](/plugins/commons-chunk-plugin), a fairly advanced tool that allows us to extract common dependencies into an existing entry chunk or an entirely new chunk. Let's use this to de-duplicate the `moment` dependency from the previous example:
+
+__webpack.config.js__
+
+``` js
+const path = require('path');
```

Comment:
Ok, so yes you are correct that maybe within the `/src` files we shouldn't but within the configs it should be ok because Node has supported it for a little while now. So I would revise my earlier statement to be Node imports within `webpack.config.js` examples should be `const name = require('package'`. That's what we're using in the rest of the docs as well and I think it's the "recommended" practice until Node supports `import` and `export` syntax.